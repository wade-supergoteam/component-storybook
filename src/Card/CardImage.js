import styled from "styled-components";
import PropTypes from "prop-types";

import { borderTopRadius, borderBottomRadius } from "../utils/border-radius";

const CardImage = styled.img`
  width: 100%; // Required because we use flexbox and this inherently applies align-self: stretch

  ${props =>
    props.position === "bottom"
      ? borderBottomRadius(props.theme.cardInnerBorderRadius)
      : borderTopRadius(props.theme.cardInnerBorderRadius)};
`;

CardImage.proptypes = {
  position: "top",
};

CardImage.proptypes = {
  position: PropTypes.oneOf(["top", "bottom"]),
};

export default CardImage;
