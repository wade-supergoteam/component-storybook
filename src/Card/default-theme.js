// @flow

export const cardSpacerY = ".75rem";
export const cardSpacerX = "1.25rem";
export const cardBorderWidth = "1px";
export const cardBorderRadius = ".25rem";
export const cardBorderColor = "rgba(0,0,0,0.125)";
export const cardInnerBorderRadius = "calc(0.25rem - 1px)";
export const cardCapBg = "rgba(0,0,0,0.03)";
export const cardBg = "#fff";
export const cardImgOverlayPadding = "1.25rem";
export const cardGroupMargin = "15px";
export const cardDeckMargin = "15px";
export const cardColumnsCount = "3";
export const cardColumnsGap = "1.25rem";
export const cardColumnsMargin = ".75rem";
