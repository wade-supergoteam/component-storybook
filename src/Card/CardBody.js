import styled from "styled-components";

import themeProp from "../utils/theme";
import { cardSpacerX } from "./default-theme";

const CardBody = styled.div`
  flex: 1 1 auto;
  padding: ${themeProp("cardSpacerX", cardSpacerX)};
`;

export default CardBody;
