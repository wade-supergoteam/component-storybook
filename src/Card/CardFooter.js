import styled from "styled-components";

import { borderRadius } from "../utils/border-radius";
import themeProp from "../utils/theme";

import { cardSpacerY, cardSpacerX, cardBorderWidth, cardBorderColor, cardInnerBorderRadius } from "./default-theme";

const CardFooter = styled.div`
  padding: ${themeProp("cardSpacerY", cardSpacerY)} ${themeProp("cardSpacerX", cardSpacerX)};
  background-color: ${props => props.theme.cardCapBg};
  border-top: ${themeProp("cardBorderWidth", cardBorderWidth)} solid ${themeProp("cardBorderColor", cardBorderColor)};

  &:last-child {
    ${borderRadius(`
      0,
      0,
      ${themeProp("cardInnerBorderRadius", cardInnerBorderRadius)},
      ${themeProp("cardInnerBorderRadius", cardInnerBorderRadius)}
    `)};
  }
`;

export default CardFooter;
