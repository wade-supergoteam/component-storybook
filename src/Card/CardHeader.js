import styled from "styled-components";

import { borderRadius } from "../utils/border-radius";
import themeProp from "../utils/theme";

import { cardSpacerY, cardSpacerX, cardCapBg, cardInnerBorderRadius, cardBorderWidth, cardBorderColor } from "./default-theme";

const CardHeader = styled.div`
  padding: ${themeProp("cardSpacerY", cardSpacerY)} ${themeProp("cardSpacerX", cardSpacerX)};
  margin-bottom: 0; // Removes the default margin-bottom of <hN>
  background-color: ${themeProp("cardCapBg", cardCapBg)};

  border-bottom: ${themeProp("cardBorderWidth", cardBorderWidth)} solid ${themeProp("cardBorderColor", cardBorderColor)}};

  &:first-child {
    ${borderRadius(`
      ${themeProp("cardInnerBorderRadius", cardInnerBorderRadius)},
      ${themeProp("cardInnerBorderRadius", cardInnerBorderRadius)},
      0,
      0`)};
  }
`;

export default CardHeader;
