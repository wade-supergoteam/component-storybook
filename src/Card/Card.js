import styled from "styled-components";
import themeProp from "../utils/theme";

import CardHeader from "./CardHeader";
import CardBody from "./CardBody";
import CardFooter from "./CardFooter";

import { borderRadius } from "../utils/border-radius";

import { cardBorderWidth, cardBorderColor, cardBorderRadius, cardBg } from "./default-theme";

const Card = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: ${themeProp("cardBg", cardBg)};
  background-clip: border-box;

  border: ${themeProp("cardBorderWidth", cardBorderWidth)} solid ${themeProp("cardBorderColor", cardBorderColor)};

  ${borderRadius(themeProp("cardBorderRadius", cardBorderRadius))};

  > hr {
    margin-right: 0;
    margin-left: 0;
  }
`;

Card.Header = CardHeader;
Card.Body = CardBody;
Card.Footer = CardFooter;

export default Card;
