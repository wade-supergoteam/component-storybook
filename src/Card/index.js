import Card from "./Card";

import CardHeader from "./CardHeader";
import CardBody from "./CardBody";
import CardFooter from "./CardFooter";
import CardImage from "./CardImage";

export { CardHeader, CardBody, CardFooter, CardImage };

export default Card;
