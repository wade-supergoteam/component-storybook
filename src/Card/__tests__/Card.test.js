import React from "react";
import renderer from "react-test-renderer";
import Card from "../";
import CardImage from "../CardImage";

test("create card", () => {
    const wrapper = renderer.create(<Card />);
    expect(wrapper).toMatchSnapshot();
});

test("With Image position bottom", () => {
    const wrapper = renderer.create(<CardImage position="bottom" />);
    expect(wrapper).toMatchSnapshot();
});

test("With Image", () => {
    const wrapper = renderer.create(<CardImage position="top" />);
    expect(wrapper).toMatchSnapshot();
});


test("Card with Header", () => {
    const wrapper = renderer.create(<Card.Header />);
    expect(wrapper).toMatchSnapshot();
});

test("Card with Header", () => {
    const wrapper = renderer.create(<Card.Footer />);
    expect(wrapper).toMatchSnapshot();
});
