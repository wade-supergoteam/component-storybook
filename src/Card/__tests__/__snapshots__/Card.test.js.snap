// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card with Header 1`] = `
<div
  className="sc-bdVaJa frOTaF"
/>
`;

exports[`Card with Header 2`] = `
<div
  className="sc-htpNat fDAeZq"
/>
`;

exports[`With Image 1`] = `
<img
  className="sc-ifAKCX iSwMuJ"
/>
`;

exports[`With Image position bottom 1`] = `
<img
  className="sc-ifAKCX gTiwhe"
/>
`;

exports[`create card 1`] = `
<div
  className="sc-bxivhb bLohfb"
/>
`;
